soh/soh/Enhancements/crowd-control/CrowdControl.cpp

```diff
@@ -187,43 +193,44 @@ void CrowdControl::ListenToServer() {
 
 void CrowdControl::ProcessActiveEffects() {
     while (isEnabled) {
-        // we only want to send events when status changes, on start we send Success,
-        // if it fails at some point, we send Pause, and when it starts to succeed again we send Success.
+        // We only want to send events when status changes, on start we send Success.
+        // If it fails at some point, we send Pause, and when it starts to succeed again we send Success.
+        // CC uses this to pause the timer on the overlay.
         activeEffectsMutex.lock();
         auto it = activeEffects.begin();
 
         while (it != activeEffects.end()) {
             Effect *effect = *it;
-            EffectResult result = ExecuteEffect(effect->type.c_str(), effect->value, false);
+            EffectResult result = CrowdControl::ExecuteEffect(effect);
+
             if (result == EffectResult::Success) {
-                // If time remaining has reached 0, we have finished the effect
+                // If time remaining has reached 0, we have finished the effect.
                 if (effect->timeRemaining <= 0) {
                     it = activeEffects.erase(std::remove(activeEffects.begin(), activeEffects.end(), effect),
                                         activeEffects.end());
-                    RemoveEffect(effect->type.c_str());
-
+                    GameInteractor::RemoveEffect(effect->giEffect);
                     delete effect;
                 } else {
-                    // If we have a success after previously being paused, fire Resume event
+                    // If we have a success after previously being paused, tell CC to resume timer.
                     if (effect->isPaused) {
                         effect->isPaused = false;
                         EmitMessage(tcpsock, effect->id, effect->timeRemaining, EffectResult::Resumed);
+                    // If not paused before, subtract time from the timer and send a Success event if
+                    // the result is different from the last time this was ran.
                     } else {
                         effect->timeRemaining -= 1000;
```

Comment:
another "not new to this PR" comment, why `1000`?

Reply:
> timed effects get put on a thread that executes every second 
might be nice to add a comment here explaining that, just having
```cpp
// timed events are put on a thread that runs once per second
effect->timeRemaining -= 1000;
```
would be enough context for someone unfamiliar with the code to understand that the 1000 is milliseconds

Reply:
Done.